src/main/java/seedu/duke/Duke.java

```diff
@@ -0,0 +1,115 @@
+package seedu.duke;
+
+import java.security.InvalidParameterException;
+import java.util.Scanner;
+
+/**
+ * Main part of logic of Duke.
+ */
+public class Duke {
+
+    /**
+     * Holds the storage of data
+     */
+    private Storage storage;
+
+    /**
+     * Manages the list of tasks
+     */
+    private TaskList taskList;
+
+    /**
+     * Parses input strings from user to task objects
+     */
+    private Parser parser;
+
+    /**
+     * Holds the user interface part
+     */
+    private Ui ui;
+
+    /**
+     * Initializes Duke object.
+     * Finishes essential settings.
+     *
+     * @param filePath Path of data file.
+     */
+    public Duke(String filePath) {
+        storage = new Storage(filePath);
+        taskList = new TaskList(storage);
+        parser = new Parser();
+        ui = new Ui();
+    }
+
+    /**
+     * Holds management of operations.
+     */
+    private void run() {
+
+        ui.welcomeWord();
+        ui.showHistory(taskList);
+
+        Scanner inputScanner = new Scanner(System.in);
+        String userInput = "";
+
+        while (!userInput.equals("bye")) {
```

Comment:
Perhaps this part can be done in the Parser class?

Reply:
Hmmm, yes, my structure still needs to optimize.